lib/kube/kubeconfig/kubeconfig.go

```diff
@@ -23,38 +26,92 @@ var log = logrus.WithFields(logrus.Fields{
 
 // Values are Teleport user data needed to generate kubeconfig entries.
 type Values struct {
-	// Name is used to name kubeconfig sections ("context", "cluster" and
+	// TeleportClusterName is used to name kubeconfig sections ("context", "cluster" and
 	// "user"). Should match Teleport cluster name.
-	Name string
+	TeleportClusterName string
 	// ClusterAddr is the public address the Kubernetes client will talk to,
 	// usually a proxy.
 	ClusterAddr string
 	// Credentials are user credentials to use for authentication the
 	// ClusterAddr. Only TLS fields (key/cert/CA) from Credentials are used.
 	Credentials *client.Key
+	// Exec contains optional values to use, when configuring tsh as an exec
+	// auth plugin in kubeconfig.
+	//
+	// If not set, static key/cert from Credentials are written to kubeconfig
+	// instead.
+	Exec *ExecValues
+}
+
+// ExecValues contain values for configuring tsh as an exec auth plugin in
+// kubeconfig.
+type ExecValues struct {
+	// TshBinaryPath is a path to the tsh binary for use as exec plugin.
+	TshBinaryPath string
+	// KubeClusters is a list of kubernetes clusters to generate contexts for.
+	// Only used when TshBinaryPath is set.
```

Comment:
This could be deleted since TshBinaryPath will be always set here since I understand only when the Exec is set (is non-nil), it will have all relevant attributes set.